gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/services/VoltageCheckService.java

```diff
@@ -29,20 +32,35 @@
 @Service
 public class VoltageCheckService {
     private final Logger businessLogger;
+    private final UrlValidationService urlValidationService;
 
-    public VoltageCheckService(Logger businessLogger) {
+    public VoltageCheckService(Logger businessLogger, UrlValidationService urlValidationService) {
         this.businessLogger = businessLogger;
+        this.urlValidationService = urlValidationService;
     }
 
     public Optional<VoltageMonitoringResult> runVoltageCheck(SweData sweData, DichotomyResult<SweDichotomyValidationData> dichotomyResult, DichotomyDirection direction) {
 
         if ((direction == DichotomyDirection.ES_FR || direction == DichotomyDirection.FR_ES) && dichotomyResult.hasValidStep()) {
             businessLogger.info("Running voltage check");
-            Crac crac = sweData.getCracFrEs().getCrac();
-            Network network = NetworkUtil.getNetworkByDirection(sweData, direction);
-            VoltageMonitoring voltageMonitoring = new VoltageMonitoring(crac, network, dichotomyResult.getHighestValidStep().getRaoResult());
-            return Optional.of(voltageMonitoring.run(LoadFlow.find().getName(), LoadFlowParameters.load(), 4));
+            try {
+                Crac crac = sweData.getCracFrEs().getCrac();
+                Network network = getNetworkWithPra(dichotomyResult);
+                VoltageMonitoring voltageMonitoring = new VoltageMonitoring(crac, network, dichotomyResult.getHighestValidStep().getRaoResult());
+                return Optional.of(voltageMonitoring.run(LoadFlow.find().getName(), LoadFlowParameters.load(), 4));
+            } catch (Exception e) {
```

Comment:
Is it possible to be more specific about the exception catching ?

Reply:
not really we are targeting robustness